cmd/process-agent/collector_test.go

```diff
@@ -345,3 +351,214 @@ func TestIgnoreResponseBody(t *testing.T) {
 		})
 	}
 }
+
+func TestCollectorRunCheckWithRealTime(t *testing.T) {
+	cfg := config.NewDefaultAgentConfig()
+	check := &mocks.CheckWithRealTime{}
```

Comment:
If I understand correctly, you should use `NewCheckWithRealTime()`, because it adds a cleanup function that actually asserts that the mocked functions were called.

Reply:
I think I was always calling AssertExpectation by hand, but I will use this way moving forward.